fastapi_asyncapi/schema.py

```diff
@@ -0,0 +1,260 @@
+
+
+# TODO: Missing other bindings!
+class OperationBinding(BaseModel):
+    http: Optional[HTTPOperationBinding]
+
+
+Bindings = List[Dict[str, OperationBinding]]
+
+
+class OperationTrait(BaseModel):
+    operationId: Optional[str]
+    summary: Optional[str]
+    tags: Optional[List[Tag]]
+    externalDocs: Optional[ExternalDocumentation]
+    bindings: Bindings
+
+
+OperationTraits = List[OperationTrait]
+
+
+class Reference(BaseModel):
+    ref: Optional[str] = Field(alias="$ref")
+
+
+class CorrelationID(BaseModel):
+    description: Optional[str]
+    location: str
+
+
+class HTTPMessageBinding(BaseModel):
+    headers: Optional[Schema]
+    bindingVersion: Union[Literal["latest"], str]
+
+
+# TODO: Make ws and kafka message binding ( 2 stage )
+class MessageBinding(BaseModel):
+    http: HTTPMessageBinding
+
+
+ExamplesMessages = List[Dict[str, Any]]
+
+
+class MessageTrait(BaseModel):
+    headers: Optional[Union[Schema, Reference]]
+    correlationId: Optional[Union[CorrelationID, Reference]]
+    schemaFormat: Optional[str]
+    contentType: Optional[str]
+    name: Optional[str]
+    title: Optional[str]
+    summary: Optional[str]
+    description: Optional[str]
+    tags: Optional[List[Tag]]
+    externalDocs: Optional[ExternalDocumentation]
+    bindings: Optional[MessageBinding]
+    examples: Optional[ExamplesMessages]
+
+
+MessagesTraits = List[MessageTrait]
+
+
+class Message(BaseModel):
+    headers: Optional[Union[Schema, Reference]]
+    payload: Optional[Any]
+    correlationId: Optional[Union[CorrelationID, Reference]]
+    schemaFormat: Optional[str]
+    contentType: Optional[str]
+    name: Optional[str]
+    title: Optional[str]
+    summary: Optional[str]
+    description: Optional[str]
+    tags: Optional[List[Tag]]
+    externalDocs: Optional[ExternalDocumentation]
+    bindings: Optional[MessageBinding]
+    examples: Optional[ExamplesMessages]
+    traits: Optional[MessagesTraits]
+
+
+class Operation(BaseModel):
+    operationId: Optional[str]
+    summary: Optional[str]
+    description: Optional[str]
+    tags: List[Tag]
+    externalDocs: Optional[ExternalDocumentation]
+    bindings: Optional[Bindings]
+    traits: Optional[OperationTraits]
+    message: Optional[Message]
+
+
+Subscribe = Operation
+Publish = Operation
+
+
+class Parameter(BaseModel):
+    description: Optional[str]
+    schema: Optional[Schema]
+    location: Optional[str]
+
+
+Parameters = List[Dict[str, Union[Parameter, Reference]]]
+
+
+ChannelBindings = Dict[str, Any]
+
+
+class ChannelItem(BaseModel):
+    ref: Optional[str] = Field(alias="$ref")
+    description: Optional[str]
+    servers: Optional[List[str]]
+    subscribe: Optional[Subscribe]
+    publish: Optional[Publish]
+    parameters: Optional[Parameters]
+    bindings: Optional[Union[ChannelBindings, Reference]]
+
+
+Security = Dict[str, List[str]]
+Channels = Dict[str, ChannelItem]
+SecurityRequirement = Dict[str, List[str]]
+ServerBinding = Dict[str, Any]
```

Comment:
https://www.asyncapi.com/docs/specifications/v2.3.0#serverBindingsObject

This should also be called `ServerBindings` instead of the singular version.
```suggestion


class ServerBindings(BaseModel):
    http: HTTPServerBinding
    # many others...
    
    class Config:
        extra = "allow"
```

Reply:
I can't to write this. Because it's not supported - https://github.com/asyncapi/bindings/tree/master/http#server-binding-object
![Screenshot from 2022-02-15 23-55-39](https://user-images.githubusercontent.com/7024995/154147841-2b332317-2b3c-4a4d-b5e2-ecb5b416f0c7.png)
e

Reply:
It says that for `ServerBinding`, but my comment says to move to `ServerBindings` :thinking:

Reply:
Yes. Should I write like this?
```python

HTTPServerBindind = Dict[str, Any]

class ServerBindings(BaseModel):
    http: HTTPServerBinding
```